lib/ecto/repo.ex

```diff
@@ -235,10 +242,36 @@ defmodule Ecto.Repo do
     * `:log` - When false, does not log the query
 
   """
+  defcallback get_by(Ecto.Queryable.t, Keyword.t, Keyword.t) :: Ecto.Model.t | nil | no_return
+
+  @doc """
+  Similar to `get_by/3` but raises `Ecto.NotSingleResult` if no record was found.
+
+  ## Options
+
+    * `:timeout` - The time in milliseconds to wait for the call to finish,
+      `:infinity` will wait indefinitely (default: 5000);
+    * `:log` - When false, does not log the query
+
+  """
+  defcallback get_by!(Ecto.Queryable.t, Keyword.t, Keyword.t) :: Ecto.Model.t | nil | no_return
+
+  @doc """
+  Fetches a single result from the query.
+
+  Returns `nil` if no result was found.
```

Comment:
Maybe it would be helpful to include an example call? Like `Repo.get_by(User, email: "foo@example.com"`